yamlfmt/yamlfmt.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/kylelemons/godebug/diff"
+	yaml "sigs.k8s.io/yaml/goyaml.v3"
+	"sigs.k8s.io/yaml/kyaml"
+)
+
+const (
+	fmtYAML  = "yaml"
+	fmtKYAML = "kyaml"
+)
+
+func main() {
+	help := flag.Bool("?", false, "print usage and exit")
+	diff := flag.Bool("d", false, "diff input files with their formatted versions")
+	write := flag.Bool("w", false, "write result to input files instead of stdout")
+	format := flag.String("o", "yaml", "output format: may be 'yaml' or 'kyaml'")
```

Comment:
I'm surprised to even offer a non-kyaml option here, let alone defaulting to it... don't we want to be opinionated towards kyaml?

Reply:
Mayber eventually but I don't KNOW that yet?  There's enough pushback that I think we should be cautious and open to being wrong.